core/src/main/scala/cats/data/Kleisli.scala

```diff
@@ -147,11 +146,10 @@ private[data] sealed abstract class KleisliInstances0 extends KleisliInstances1
   implicit def catsDataMonadForKleisliId[A]: Monad[Kleisli[Id, A, ?]] =
     catsDataMonadForKleisli[Id, A]
 
-  implicit def catsDataCommutativeArrowForKleisli[F[_]](implicit M: CommutativeMonad[F]): CommutativeArrow[Kleisli[F, ?, ?]] =
-    new KleisliCommutativeArrow[F] {def F: CommutativeMonad[F] = M }
-
-  implicit val catsDataCommutativeArrowForKleisliId: CommutativeArrow[Kleisli[Id, ?, ?]] =
-    catsDataCommutativeArrowForKleisli[Id]
+  implicit def catsDataArrowForKleisli[F[_]](implicit M: Monad[F]): Arrow[Kleisli[F, ?, ?]] =
```

Comment:
If I am not mistaken, since both `Arrow` and `ArrowChoice` instances have the same requirement (`Monad[F]`), one `ArrowChoice` instance should be fine. You can basically leave `catsDataCommutativeArrowForKleisliId` and `catsDataCommutativeArrowForKleisli` as is.